AzureMonitorAgent/agent.py

```diff
@@ -949,8 +956,32 @@ def set_proxy(address, username, password):
         run_command_and_log("systemctl daemon-reload")
         
     except:
-        log_and_exit("enable", MissingorInvalidParameterErrorCode, "Failed to update /etc/systemd/system/azuremonitor-coreagent.service.d and mkdir -p /etc/systemd/system/metrics-extension.service.d" )
+        log_and_exit("enable", MissingorInvalidParameterErrorCode, "Failed to update /etc/systemd/system/azuremonitor-coreagent.service.d and /etc/systemd/system/metrics-extension.service.d" )
+
+def unset_proxy():
+    """
+    # Unset proxy http_proxy env var in dependent services
+    """
     
+    try:
+        hasSettings=False
+        
+        # Update Coreagent
+        if os.path.exists("/etc/systemd/system/azuremonitor-coreagent.service.d/proxy.conf"):
+            os.remove("/etc/systemd/system/azuremonitor-coreagent.service.d/proxy.conf")
+            hasSettings=True
+            
+        # Update ME
+        if os.path.exists("/etc/systemd/system/metrics-extension.service.d/proxy.conf"):
+            os.remove("/etc/systemd/system/metrics-extension.service.d/proxy.conf")
+            hasSettings=True
+            
+        if hasSettings:
+            run_command_and_log("systemctl daemon-reload")
```

Comment:
this requires a service restart as well after daemon-reload -- are we doing that later after this gets called? (I assume so, just double checking)

Reply:
we were previously not doing it but it worked with just daemon-reload but I have explicitly added it as well now